pynos/versions/ver_7/ver_7_1_0/nsx.py

```diff
@@ -0,0 +1,197 @@
+# Copyright 2017 Great Software Laboratory Pvt. Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+import logging
+import re
+import xml.etree.ElementTree as ET
+from ipaddress import ip_interface
+import pynos.utilities
+from ipaddress import ip_interface
+from pynos.versions.ver_7.ver_7_1_0.yang.brocade_tunnels import brocade_tunnels
+
+
+
+class Nsx():
+    """NSX class containing all NSX related methods and
+    attributes.
+    """
+    def __init__(self,callback):
+        """
+        NSX init function
+
+        Args:
+            callback: Callback function that will be called for each action
+
+        Returns:
+            NSX Object
+
+        Raises:
+            None
+        """
+        self._callback = callback
+        self._brocade_tunnels = brocade_tunnels (callback=pynos.utilities.return_xml)
+
+    def nsx_controller_name(self,**kwargs):
```

Comment:
In our case we will be configuring factory reset switches, therefore we did not find any requirement to delete controller, but if required we will add it in future.